library/src/test/java/com/getstream/sdk/chat/models/ChannelComparisonTest.java

```diff
@@ -0,0 +1,89 @@
+package com.getstream.sdk.chat.models;
+
+import com.getstream.sdk.chat.model.Channel;
+import com.getstream.sdk.chat.rest.core.Client;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ChannelComparisonTest {
+
+    private static String TEST_CHANNEL_TYPE = "testChannelType";
+
+    private static String TEST_CHANNEL_ID_1 = "testChannelId_1";
+    private static String TEST_CHANNEL_ID_2 = "testChannelId_2";
+    private static String TEST_CHANNEL_ID_3 = "testChannelId_3";
+
+    @Mock
+    Client client;
+
+    private Channel x, y, z, x_clone;
+
+    @BeforeEach
+    void initTest() {
+        MockitoAnnotations.initMocks(this);
+
+        x = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID_1);
+        x_clone = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID_1);
+        y = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID_2);
+        z = new Channel(client, TEST_CHANNEL_TYPE, TEST_CHANNEL_ID_3);
+    }
+
+
+    @Test
+    void equalsTest() {
```

Comment:
- tests should be easy to read
- test 1 thing with 1 test
- extract common test code if needed
- test names should make sense (easier when you only test 1 thing)

```
@Test
void equalsIsTransitive() {
    assertTrue(a.equals(b) && b.equals(c) && a.equals(a));
}

@Test
void equalsIsReflexive() {
    assertTrue(a.equals(a));
}

@Test
void equalsIsSymmetric() {
    // with a and be equal
    assertEqual(a.equals(b), b.equals(a));

    // with a and be not equal
    assertEqual(a.equals(b), b.equals(a));
}
```